azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -1375,3 +1408,45 @@ func resourceArmContainerGroupPortsHash(v interface{}) int {
 
 	return hashcode.String(buf.String())
 }
+
+func flattenContainerGroupDnsConfig(input *containerinstance.DNSConfiguration) []interface{} {
+	if input == nil {
+		return nil
+	}
+	outputArr := make([]interface{}, 1)
+	output := make(map[string]interface{})
+
+	if v := input.SearchDomains; v != nil {
```

Comment:
If the field is absent in `input` here, can we assign a zero value of that type into the map as other resources' flatten function does?

(same applies to other properties below)

Reply:
I'm assuming, but could be wrong here that if a pointer to a string that is not nil will be initialized to its zero value automatically. So I think by putting `output["nameservers"] = *v` when I dereference the non-nil pointer I _at least_ get `""` and at most I get some non-empty value. Am I correct? I'm still learning go :)

Reply:
You're right. While what I suggested is to do something like:

```go
searchDomains := ""
if input.SearchDomains != nil {
    searchDomains = *input.SearchDomains
}
output["search_domains"] = searchDomains
```

The intent is to always set `search_domains`.

Reply:
Please check I believe I have resolved

Reply:
Sure.